src/features/world/lib/AudioController.ts

```diff
@@ -0,0 +1,77 @@
+import Phaser from "phaser";
+
+export type Sound =
+  | Phaser.Sound.HTML5AudioSound
+  | Phaser.Sound.WebAudioSound
+  | Phaser.Sound.NoAudioSound;
+
+export class WalkAudioController {
+  // private music: Sound;
+  private walkSound: Sound;
+
+  constructor(walkSound: Sound) {
+    // this.music = music;
+    this.walkSound = walkSound;
+  }
+
+  handleWalkSound(isWalking: boolean): void {
+    if (isWalking) {
+      if (!this.walkSound.isPlaying) {
+        this.walkSound.play({ loop: true, volume: 0.05, rate: 0.62 });
+      }
+    } else {
+      if (this.walkSound.isPlaying) {
+        this.walkSound.stop();
+      }
+    }
+  }
+}
+
+export type AudioProps = {
+  sound: Sound;
+  distanceThreshold: number;
+  coordinates: {
+    x: number;
+    y: number;
+  };
+  maxVolume: number;
+};
+
+export class AudioController {
+  // private music: Sound;
+  private audio: AudioProps;
+
+  constructor(audio: AudioProps) {
+    this.audio = audio;
+  }
+
+  setVolumeAndPan(playerX: number, playerY: number): void {
+    const distanceToSoundSource = Phaser.Math.Distance.Between(
+      playerX,
+      playerY,
+      this.audio.coordinates.x,
+      this.audio.coordinates.y
+    );
+
```

Comment:
When we have a lot of spatial sounds, it could be useful to early return here to avoid running all of the computational logic below on each update.

For instance, if player is no where near the sound don't worry about adjusting it.